.github/workflows/commit-format.yml

```diff
@@ -0,0 +1,43 @@
+name: GNU Commit Format Checker
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_commit_formats:
+    runs-on: ubuntu-latest
+    name: Commits Check
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Install Deps
+        run: |
+          sudo apt-get update;
+          sudo apt-get install -y \
+            python3 \
+            python3-git
+      
+      - name: Get PR Commits
+        run: |
+          git rev-list \
+              origin/master..${{ github.event.pull_request.head.sha }} \
+              > commits.txt
+
+      - name: Check commits
+        run: |
+          for i in `cat commits.txt`; do \
+              echo "Checking commit $i" \
+              python3 contrib/gcc-changelog/git_check_commit.py \
+                  | tee results.log ; \
+          done; \
+          if grep -e "ERR:" -e "FAILED" results.log; \
+          then \
+              echo "::error commit's are not formatted correctly"; \
+              exit 1; \
+          fi
```

Comment:
I've noticed that you can directly give a rev-list to `contrib/gcc-changelog/git_check_commit.py`, so doing `contrib/gcc-changelog/git_check_commit.py origin/master..${{ github.event.pull_request.head.sha }}` is enough.

But then I'm not sure about the output and how it would all look. If it was possible to paste the output of that command into a github comment on the issue that'd be fantastic. But for now I think just running the above comments and having the contributor check the CI job is enough

Reply:
```
Checking b22634281fff352fcf71dd4fbbf6e6fcbc9a46cf: OK
Checking fe67b75f59c64a9b7cd3b997d406b986c2095cf6: OK
Checking 9cfa7d7e3c3d7144a7b6079e7dc65a38a5327e70: OK
Checking 3b4c6e67710b09beccb7d0acf7f7257564d8a6f5: OK
Checking 39a7b603380c6f4383357a6ae1d6c516dc677f29: OK
Checking b7ed6c43a80e06082baad5336be0fa943a878d40: OK
Checking 00fdfe9de2e2205330f312364b3b60fca3005b61: OK
Checking 1f038e845bbdeae9dddf1810fb3e6c9ad1b79f13: OK
Checking e59cd0db822e325868128281a81ee356a6914f52: OK
Checking 24993939b7e405cdb8fa7ab6b388913a5d9fb6ce: OK
Checking 8daf80ff0ecd2aee50bf8e4f0f0dda906aeb190d: OK
```
and when it's not ok:
```
Checking 7ccb3122c9de5548398b6165890b912b263fe351: OK
Checking 1d2276d1bdb2f25f03ccc1078fbcfe059eb5409e: OK
Checking 616706e44e886eb447a88b0ea99df5b7c38eed88: FAILED
ERR: cannot find a ChangeLog location in message
Checking ff15b6c1dc22d86cb0cc1af162591f03d84771c5: OK
Checking 66c47521eac4e042880f63f731422112e4b1994e: FAILED
ERR: unchanged file mentioned in a ChangeLog (did you mean "gcc/ada/exp_ch3.adb"?): "gcc/ada/foo.adb"
Checking b7ed6c43a80e06082baad5336be0fa943a878d40: OK
Checking 00fdfe9de2e2205330f312364b3b60fca3005b61: OK
Checking 1f038e845bbdeae9dddf1810fb3e6c9ad1b79f13: OK
Checking e59cd0db822e325868128281a81ee356a6914f52: OK
Checking 24993939b7e405cdb8fa7ab6b388913a5d9fb6ce: OK
Checking 8daf80ff0ecd2aee50bf8e4f0f0dda906aeb190d: OK
```

Reply:
Yes, I meant the output in Github sorry :) I've had to face the wrath of `git_check_commit.py` multiple times now haha